hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -0,0 +1,1009 @@
+        val aggSig = agg_signature(it)
+        val i = ir_value_expr(env)(it)
+        val args = ir_value_exprs(env)(it)
+        SeqOp(i, args, aggSig)
+      case "Begin" =>
+        val xs = ir_value_children(env)(it)
+        Begin(xs)
+      case "MakeStruct" =>
+        val fields = named_value_irs(env)(it)
+        MakeStruct(fields)
+      case "SelectFields" =>
+        val fields = identifiers(it)
+        val old = ir_value_expr(env)(it)
+        SelectFields(old, fields)
+      case "InsertFields" =>
+        val old = ir_value_expr(env)(it)
+        val fields = named_value_irs(env)(it)
+        InsertFields(old, fields)
+      case "GetField" =>
+        val name = identifier(it)
+        val s = ir_value_expr(env)(it)
+        GetField(s, name)
+      case "MakeTuple" =>
+        val args = ir_value_children(env)(it)
+        MakeTuple(args)
+      case "GetTupleElement" =>
+        val idx = int32_literal(it)
+        val tuple = ir_value_expr(env)(it)
+        GetTupleElement(tuple, idx)
+      case "StringSlice" =>
+        val s = ir_value_expr(env)(it)
+        val start = ir_value_expr(env)(it)
+        val end = ir_value_expr(env)(it)
+        StringSlice(s, start, end)
+      case "StringLength" =>
+        val s = ir_value_expr(env)(it)
+        StringLength(s)
+      case "In" =>
+        val typ = type_expr(it)
+        val idx = int32_literal(it)
+        In(idx, typ)
+      case "Die" =>
+        val typ = type_expr(it)
+        val msg = string_literal(it)
+        Die(msg, typ)
+      case "ApplySeeded" =>
+        val function = identifier(it)
+        val seed = int64_literal(it)
+        val args = ir_value_children(env)(it)
+        ApplySeeded(function, args, seed)
+      case "ApplyIR" | "ApplySpecial" | "Apply" =>
+        val function = identifier(it)
+        val args = ir_value_children(env)(it)
+        invoke(function, args: _*)
+      case "Uniroot" =>
+        val name = identifier(it)
+        val function = ir_value_expr(env + (name -> TFloat64()))(it)
+        val min = ir_value_expr(env)(it)
+        val max = ir_value_expr(env)(it)
+        Uniroot(name, function, min, max)
+      case "TableCount" =>
+        val child = table_ir(env)(it)
+        TableCount(child)
+      case "JavaIR" =>
+        val name = identifier(it)
+        env.irMap(name).asInstanceOf[IR]
+    }
+  }
+
+  def table_irs(env: IRParserEnvironment)(it: TokenIterator): Array[TableIR] = {
+    punctuation(it, "(")
+    val tirs = table_ir_children(env)(it)
+    punctuation(it, ")")
+    tirs
+  }
+
+  def table_ir_children(env: IRParserEnvironment)(it: TokenIterator): Array[TableIR] =
+    repUntil(it, table_ir(env), PunctuationToken(")"))
+
+  def table_ir(env: IRParserEnvironment)(it: TokenIterator): TableIR = {
+    punctuation(it, "(")
+    val ir = table_ir_1(env)(it)
+    punctuation(it, ")")
+    ir
+  }
+
+  def table_ir_1(env: IRParserEnvironment)(it: TokenIterator): TableIR = {
+    // FIXME TableImport
```

Comment:
Fix you.

Reply:
I just ported the FIXME message from the expr.Parser file.